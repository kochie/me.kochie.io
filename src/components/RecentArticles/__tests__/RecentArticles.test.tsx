import { act, create, ReactTestRenderer } from 'react-test-renderer'
import * as React from 'react'
import { jest } from '@jest/globals'
// import RecentArticles from '../RecentArticles'
// import { SWRConfig } from 'swr'

// global.fetch = jest.fn(() => Promise.resolve({
//     json: () => Promise.resolve(MOCK_DATA)
// }));

describe('Recent Articles', () => {
  const MOCK_DATA = {
    items: [
      {
        id: 'https://blog.kochie.io/articles/02-contact-tracing',
        url: 'https://blog.kochie.io/articles/02-contact-tracing',
        title: 'Building a Contact Tracing Platform',
        summary:
          "Learn how to automate and build an application to trace exposures using Amazon's Cloud Development Kit (CDK)",
        image:
          'https://blog.kochie.io/articles-1/02-contact-tracing/omar-flores-MOO6k3RaiwE-unsplash.jpg',
        date_modified: '2021-08-22T23:00:00.000Z',
        author: {
          name: 'Robert Koch',
          url: 'https://blog.kochie.io/authors/kochie',
        },
      },
    ],
  }

  beforeAll(async () => {
    jest.unstable_mockModule('swr', () => {
      // console.log('mock')
      return {
        default: jest.fn(() => ({
          data: MOCK_DATA,
          error: null,
        })),
      }
    })

    await import('swr')
  })

  it('recent articles renders correctly', async () => {
    const RecentArticles = await import('../RecentArticles')
    let tree: ReactTestRenderer

    tree = create(<RecentArticles.default />)

    expect(tree.toJSON()).toMatchSnapshot()
  })
})
