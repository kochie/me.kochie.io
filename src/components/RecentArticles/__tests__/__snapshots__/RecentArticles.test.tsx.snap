// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recent articles renders correctly 1`] = `
<div>
  <h2
    className="text-xl mb-6"
  >
    Recent Articles
  </h2>
  <div
    className="flex flex-col gap-8"
  >
    <div
      className="group flex"
    >
      <div
        className="overflow-hidden rounded-md flex-none relative h-40 w-80"
      >
        <a
          href="https://blog.kochie.io/articles/02-contact-tracing"
        >
          <div
            style={
              Object {
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "left": 0,
                "margin": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <noscript>
              <img
                alt="Building a Contact Tracing Platform"
                className="group-hover:scale-110 transition transform-gpu duration-100"
                data-nimg={true}
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=https%3A%2F%2Fblog.kochie.io%2Farticles-1%2F02-contact-tracing%2Fomar-flores-MOO6k3RaiwE-unsplash.jpg&w=1200&q=75"
                srcSet="/_next/image?url=https%3A%2F%2Fblog.kochie.io%2Farticles-1%2F02-contact-tracing%2Fomar-flores-MOO6k3RaiwE-unsplash.jpg&w=320&q=75 320w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Farticles-1%2F02-contact-tracing%2Fomar-flores-MOO6k3RaiwE-unsplash.jpg&w=420&q=75 420w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Farticles-1%2F02-contact-tracing%2Fomar-flores-MOO6k3RaiwE-unsplash.jpg&w=768&q=75 768w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Farticles-1%2F02-contact-tracing%2Fomar-flores-MOO6k3RaiwE-unsplash.jpg&w=1024&q=75 1024w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Farticles-1%2F02-contact-tracing%2Fomar-flores-MOO6k3RaiwE-unsplash.jpg&w=1200&q=75 1200w"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": "cover",
                    "objectPosition": "center",
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
            <img
              alt="Building a Contact Tracing Platform"
              className="group-hover:scale-110 transition transform-gpu duration-100"
              data-nimg={true}
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "backgroundImage": "url(\\"data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAMCAgMCAgMDAwMEAwMEBQgFBQQEBQoHBwYIDAoMDAsKCwsNDhIQDQ4RDgsLEBYQERMUFRUVDA8XGBYUGBIUFRT/2wBDAQMEBAUEBQkFBQkUDQsNFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBT/wAARCAAKABIDASIAAhEBAxEB/8QAGQAAAgMBAAAAAAAAAAAAAAAAAAYEBQcI/8QAKBAAAQMDAwMDBQAAAAAAAAAAAQIDBBEFABIhBgcjMRQiEzJCUXGR/8QAFwEAAwEAAAAAAAAAAAAAAAAAAAIEBf/EABwRAAEEAwEAAAAAAAAAAAAAAAEAQQIhE9HwA//aAAwDAQACEQMRAD8AaLZceON9VUT4txmRum1FMszi6ta3wNtKSRXcmtCk/UP3lnJ5d0yb6d8kXeEwWJchC2rEw++4j1CiSEgJAAVt7qHb20ziPgFwlVvSfUvaWk1QPkNEdsePxk2Dc5l4F4M+W/OKEVSZLhc00QkClSabE/3FHnKJMqdloEihbOtEcfiuOLWu4nWo1PeT5wxFXHa1q7aPJ+0YZPjHE7RkPAaX/9k=\\")",
                  "backgroundPosition": "center",
                  "backgroundSize": "cover",
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "filter": "blur(20px)",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": "center",
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </div>
        </a>
      </div>
      <div
        className="ml-5"
      >
        <h3
          className="hover:underline"
        >
          <a
            href="https://blog.kochie.io/articles/02-contact-tracing"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Building a Contact Tracing Platform
          </a>
        </h3>
        <p>
          Learn how to automate and build an application to trace exposures using Amazon's Cloud Development Kit (CDK)
        </p>
      </div>
    </div>
  </div>
</div>
`;
